src/core_cli/Core_command.ml

```diff
@@ -188,8 +188,9 @@ let semgrep_core_with_one_pattern (config : Core_scan_config.t) : unit =
       let files =
         target_info
         |> List_.map (function
-             | `CodeTarget (t : Input_to_core_t.code_target) -> t.path
-             | `LockfileTarget (t : Input_to_core_t.lockfile_target) -> t.path)
+               | Target_location.Code { file; _ }
+               | Lockfile { file; _ }
+               -> Fpath.to_string file)
```

Comment:
1. is done in 135dff8da0424f9a79b39d7a72e2c7c10936e403

Reply:
Re: 1. See my comment [here](https://github.com/semgrep/semgrep/pull/9715/files/2193eb567fd8c8fc924cb989e1726c2602275a44#r1480677358). This should use the function `Target_location.file` however; so I have made that update in 4014a2d3f0523b52e9be25b1ceab1123c3b60397.

Reply:
3. I do not think this can easily be done here. `file` must be a string for a later call to `Core_error.try_with_print_exn_and_reraise` or `Core_error.try_with_exn_to_error`. Each function calls `Core_error.exn_to_error`. That function contains a check

```ocaml
          let loc =
            (* TODO: we shouldn't build Tok.t w/out a filename, but
               lets do it here so we don't crash until we do *)
            if not String.(equal file "") then Tok.first_loc_of_file file
            else Tok.fake_location
```
which to me implies that `file` may be an empty string by this point. However, if file were instead `Fpath.t`, [we cannot create such a value](https://erratique.ch/software/fpath/doc/Fpath/index.html#val-of_string) generally. At this usage, since the values have already been validated to be valid paths, I think that would be fine, but it would potentially pose issues at other uses of `Core_error.exn_to_error` or functions which call it.